go/logictest/runner.go

```diff
@@ -246,11 +246,43 @@ func runTestFile(harness Harness, file string) {
 	}
 }
 
+type R struct {
+	schema string
+	results []string
+	cont bool
+	err error
+}
+
 // Executes a single record and returns whether execution of records should continue
 func executeRecord(harness Harness, record *parser.Record) (schema string, results []string, cont bool, err error) {
-	startTime = time.Now()
 	currRecord = record
 
+	tc := make(chan *R, 1)
+	go func() {
+		schema, results, cont, err := execute(harness, record)
+		tc <- &R{
+			schema: schema,
+			results: results,
+			cont: cont,
+			err: err,
+		}
+	}()
+
+	select {
+	case res := <-tc:
+		return res.schema, res.results, res.cont, res.err
+	case <-time.After(1 * time.Hour):
+		logTimeout("Executing query took too long")
+
+		// set cont to false so that no subsequent records in this file
+		// we assume they will likely also timeout
+		return "", []string{}, false, nil
```

Comment:
Ideally we would return a custom timeout error here, so that the caller can automatically mark all the remaining tests in the file as timed out.